File: python/bc2686.py
res = ""
while True:
    try:
        an = float(input())

        if an < 0 or an > 360:
            h = 0
            m = 0
            s = 0
        else:
            h = (an // 15)
            m = (an % 15) * 4
            s = round((m - int(m)) * 60)

            h = (h + 6) % 24

        if an < 90 or an == 360:
            res = "Bom Dia!!"
        elif an < 180:
            res = "Boa Tarde!!"
        elif an < 270:
            res = "Boa Noite!!"
        else:
            res = "De Madrugada!!"

        print(res)
        print("{:02.0f}:{:02.0f}:{:02.0f}".format(h, m, s))
    except EOFError:
        break

File: c99/bc2686.c
#include <stdio.h>
#include <math.h>
#include <string.h>

int main()
{
	double an, h, m, s;
	char res[20];
	
	while(1)
	{
		if(scanf("%lf", &an) == 1)
		{
			h = an / 15;
			m = fmod(an ,15) * 4;
			s = round((m - floor(m)) * 60);
			
			if(an < 90 || an == 360) strcpy(res, "Bom Dia!!");
			else if(an < 180) strcpy(res, "Boa Tarde!!");
			else if(an < 270) strcpy(res, "Boa Noite!!");
			else strcpy(res, "De Madrugada!!");
			
			h = fmod(h + 6, 24);
			puts(res);
			printf("%02.0f:%02.0f:%02.0f\n", h, m, s);
		}
		else break;
	}
}
